File: C#/sigmf/sigmf/ext/global/waveform/Waveform.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace sigmf
{
    [Serializable()]
    public class Waveform : IEEE80211p
    {
        private static readonly long serialVersionUID = 1L;

        protected Dictionary<string, Object> otherfields = new Dictionary<string, Object>();

        public Dictionary<string, object> getOtherFields()
        {
            return otherfields;
        }
        public void add(string key, object value)
        {
            otherfields.Add(key, value);
        }
    }
}

File: java/src/main/java/gov/doc/ntia/sigmf/ext/waveform/Waveform.java
package gov.doc.ntia.sigmf.ext.waveform;

import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.annotation.JsonSubTypes;
import com.fasterxml.jackson.annotation.JsonTypeInfo;
import gov.doc.ntia.sigmf.ext.waveform.Ieee80211p;
import java.io.Serializable;

import javax.validation.constraints.NotNull;

@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonTypeInfo(use = JsonTypeInfo.Id.NAME, include = JsonTypeInfo.As.PROPERTY, property = "model")
@JsonSubTypes({@JsonSubTypes.Type(value = Ieee80211p.class, name = "IEEE80211p")})
public class Waveform implements Serializable {

  private static final long serialVersionUID = 1L;

  @NotNull
  @JsonProperty(value = "model", required = true)
  protected String model;

  @JsonProperty(value = "description", required = false)
  protected String description;

  public String getModel() {
    return model;
  }

  public void setModel(String model) {
    this.model = model;
  }

  public String getDescription() {
    return description;
  }

  public void setDescription(String description) {
    this.description = description;
  }
}
